src/vs/workbench/contrib/searchEditor/browser/searchEditorModel.ts

```diff
@@ -36,20 +37,36 @@ export class SearchEditorModel {
 		@IModeService private readonly modeService: IModeService) {
 		this.onModelResolved = new Promise<ITextModel>(resolve => this.resolveContents = resolve);
 		this.onModelResolved.then(model => this.cachedContentsModel = model);
-		this.ongoingResolve = workingCopyBackupService.resolve({ resource: modelUri, typeId: NO_TYPE_ID })
-			.then(backup => {
-				const model = modelService.getModel(modelUri);
-				if (model) {
-					return model;
+
+		this.ongoingResolve = (async () => {
+			let discardLegacyBackup = false;
+			let backup = await workingCopyBackupService.resolve({ resource: modelUri, typeId: SearchEditorWorkingCopyTypeId });
+			if (!backup) {
+				// TODO@bpasero remove this fallback after some releases
+				backup = await workingCopyBackupService.resolve({ resource: modelUri, typeId: NO_TYPE_ID });
+
+				if (backup && modelUri.scheme === SearchEditorScheme) {
```

Comment:
I believe the modelUri will always be search-editor scheme

Reply:
Yeah I just wanted to be 100% sure I discard a backup from the search editor domain, it is overly protective.